# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1473991814 14400
#      Thu Sep 15 22:10:14 2016 -0400
# Node ID 1d3b7153053900499cdc8b6aac7061f51155f686
# Parent  dbe0ec8d53b55e48ee1d8a5d67d3cb59176ba4c3
fix compilation of gl-render.cc if OpenGL is missing

* gl-render.cc (opengl_renderer::draw): Avoid calling OpenGL functions
if OpenGL is missing.
(opengl_renderer::set_linestyle, opengl_renderer::render_grid):
Mark linewidth as unused if OpenGL is missing.

diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -692,20 +692,24 @@ namespace octave
           draw_uibuttongroup (dynamic_cast<const uibuttongroup::properties&> (props), go);
       }
     else
       {
         warning ("opengl_renderer: cannot render object of type '%s'",
                  props.graphics_object_name ().c_str ());
       }
 
+#if defined (HAVE_OPENGL)
+
     GLenum gl_error = glGetError ();
     if (gl_error)
       warning ("opengl_renderer: Error %d occurred drawing '%s' object",
                gl_error, props.graphics_object_name ().c_str ());
+
+#endif
   }
 
 #if defined (HAVE_OPENGL)
 
   static std::string
   gl_get_string (GLenum id)
   {
     // This is kind of ugly, but glGetString returns a pointer to GLubyte
@@ -896,16 +900,17 @@ namespace octave
       }
     glEnd ();
     set_linestyle ("-");  // Disable LineStipple
     double black[3] = {0, 0, 0};
     glColor3dv (black);
 
 #else
 
+    octave_unused_parameter (linewidth);
     octave_unused_parameter (gridstyle);
     octave_unused_parameter (gridcolor);
     octave_unused_parameter (gridalpha);
     octave_unused_parameter (ticks);
     octave_unused_parameter (lim1);
     octave_unused_parameter (lim2);
     octave_unused_parameter (p1);
     octave_unused_parameter (p1N);
@@ -3633,16 +3638,17 @@ namespace octave
       glDisable (GL_LINE_STIPPLE);
     else
       glEnable (GL_LINE_STIPPLE);
 
 #else
 
     octave_unused_parameter (s);
     octave_unused_parameter (use_stipple);
+    octave_unused_parameter (linewidth);
 
     // This shouldn't happen because construction of opengl_renderer
     // objects is supposed to be impossible if OpenGL is not available.
 
     panic_impossible ();
 
 #endif
   }
